packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+/// icons.
+///
+/// This widget represents a leaf entry in a menu that is part of a [MenuBar].
+/// It shows a hint for an associated shortcut, if any. When selected via click,
+/// hitting enter while focused, or activating the associated [shortcut], it
+/// will call its [onSelected] callback or fire its [onSelectedIntent] intent,
+/// depending on which is defined. If neither is defined, then this item will be
+/// disabled.
+///
+/// {@macro flutter.material.menu_bar.shortcuts_note}
+///
+/// See also:
+///
+/// * [MenuBar], a class that creates a top level menu bar in a Material Design
+///   style.
+/// * [MenuButton], a menu item which manages a submenu.
+/// * [MenuItemGroup], a menu item which collects its members into a group
+///   separated from other menu items by a divider.
+/// * [MenuItemButton], a leaf menu item which displays the label, an optional
+///   shortcut label, and optional leading and trailing icons.
+/// * [createCascadingMenu], a function that creates a [MenuEntry] that allows
+///   creation and management of a cascading menu anywhere.
+/// * [MenuController], a class that allows controlling and connecting menus.
+/// * [PlatformMenuBar], which creates a menu bar that is rendered by the host
+///   platform instead of by Flutter (on macOS, for example).
+/// * [ShortcutRegistry], a registry of shortcuts that apply for the entire
+///   application.
+/// * [VoidCallbackIntent] to define intents that will call a [VoidCallback] and
+///   work with the [Actions] and [Shortcuts] system.
+/// * [CallbackShortcuts] to define shortcuts that simply call a callback and
+///   don't involve using [Actions].
+/// * [PlatformMenuBar], a class that renders similar menu bar items from a
+///   [PlatformMenuItem] using platform-native APIs.
+class MenuItemButton extends StatefulWidget {
+  /// Creates a const [MenuItemButton].
+  ///
+  /// The [label] attribute is required.
+  const MenuItemButton({
+    super.key,
+    this.shortcut,
+    this.onSelected,
+    this.onSelectedIntent,
+    this.onHover,
+    this.focusNode,
+    this.leadingIcon,
+    this.trailingIcon,
+    this.semanticsLabel,
+    this.backgroundColor,
+    this.foregroundColor,
+    this.overlayColor,
+    this.textStyle,
+    this.padding,
+    this.shape,
+    required this.label,
+  }) : assert(onSelected == null || onSelectedIntent == null,
+            'Only one of onSelected or onSelectedIntent may be specified');
+
+  /// A required widget displaying the label for this item in the menu.
+  final Widget label;
+
+  /// The optional shortcut that selects this [MenuItemButton].
+  ///
+  /// This shortcut is only enabled when [onSelected] is set.
+  final MenuSerializableShortcut? shortcut;
+
+  /// The function called when the mouse leaves or enters this menu item's
+  /// button.
+  final ValueChanged<bool>? onHover;
+
+  /// Returns a callback, if any, to be invoked if the platform menu receives a
```

Comment:
"Returns a callback, if any, to be invoked" could be A callback that is invoked

Same comment about the onSelectedIntent property